### PR TITLE
OOP school library: preserve data

### PR DESCRIPTION
**Pull request by BahirHakimy**
## In this branch we have done the following:
- Created `Storage` class to deal with reading and writing files
- Created `LibraryStorage` class to handle serialization, saving, and loading of the data
- Added exception handling to the mentioned files

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -33,7 +39,7 @@ def main
     handle_option(option, operations) if option.to_i.between?(1, 6)
     exit = true if option == '7'
   end
-  puts 'Thank you for using this app!'
+  goodBye
 end
```
Comment (AmaduKamara) [inline main.rb]: - [ ] Good job so far on your project, but when I enter option 7 to close the app so the entered data are preserved, the app crashes with the error below and I'm not able to know whether the data are stored successfully or not. Kindly fix that 👍🏽
- [ ] Please be sure you test the app and everything works fine before you submit for a review 👍🏽

Screenshot 2023-05-31 at 7 54 57 PM

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️ ♻️ ♻️

Hi Team,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✅ You used Gitflow correctly 👍🏽
✅ The Linter checks passed successfully 👍🏽
✅ The UI looks great 👍🏽

## Required Changes ♻️
_Check the comments under the review._

- [ ] Currently, the entered data are unable to be preserved and the app crashes upon exit. Kindly work on the error shown in the review feedback so the app closes as expected and the data are preserved in their separate .json files 👍🏽

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: main.rb (hunk 1 above)
Comment (BahirHakimy) [reply main.rb]: Thanks @AmaduKamara, We didn't notice that, I will make the necessary changes

**Review by OyePriscilla — APPROVED** [review]
Hi @BahirHakimy and @kkolade 👏 👏, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
App preserved data on exit ✔️ 
App loaded data on start-up ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by OyePriscilla — APPROVED** [review]
Hi @BahirHakimy and @kkolade 👏 👏, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
App preserved data on exit ✔️ 
App loaded data on start-up ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**